### ISSUE TITLE
Finished Rewards processing loop before rewards-end-block 

### ISSUE THREAD

**Issue by danbryan** [DefiantLabs/cosmos-tax-cli#348]
@KyleMoser  do you mind taking a long at this.  I think it's one of the last issues around rewards indexing.

when using 
```
    rewards-start-block = 1
    rewards-end-block = 2612736
```
I get this, see attached
[rewards.txt](https://github.com/DefiantLabs/cosmos-tax-cli/files/11069337/rewards.txt)

Points of note:

- it doesn't look like it searches block 1 - 13363 for rewards, not sure if this matters, but i think there should be a way to tell the code to index sequentially, if needed.
- It hit the Finished Rewards processing loop, before it got to block 2612736.  
- It didn't find any rewards in the 36 minutes it successfully ran for

**Comment by KyleMoser** [comment]
1. I fixed the start height so it will be honored if set explicitly in the config.
2. I fixed the for loop condition so it will stop at the correct block height (I think...I tested with 1-200). The new condition in question is `startHeight+delta <= endHeight`. 

I don't think point 3 "It didn't find any rewards in the 36 minutes it successfully ran for" really matters:

- To find Osmosis rewards epochs (which only happen once daily at predictable intervals), the indexer does a smart search. Based on the last known delta (number of blocks between two previous epochs) it will "jump" that many blocks to find the next epoch. 
- It does this by searching backwards/forwards simultaneously, and giving up if an epoch isn't found within 1000 blocks in either direction. Since you're starting at block one it doesn't know of any previous epochs, and therefore will do a block by block sequential search to find the next epoch. So I don't think the 36 minute "no rewards found" result is indicative of any issue.
